### PR TITLE
Fix: threesome joins with parallel replicas

### PR DESCRIPTION
**Pull request by devcrafter**
### Changelog category (leave one):
- Not for changelog (changelog entry is not required)

> Information about CI checks: https://clickhouse.com/docs/en/development/continuous-integration/

#### CI Settings (Only check the boxes if you know what you are doing):
- [ ]  Allow: All Required Checks
- [ ]  Allow: Stateless tests
- [ ]  Allow: Stateful tests
- [ ]  Allow: Integration Tests
- [ ]  Allow: Performance tests
- [ ]  Allow: All Builds
- [ ]  Allow: batch 1, 2 for multi-batch jobs
- [ ]  Allow: batch 3, 4, 5, 6 for multi-batch jobs
---
- [ ]  Exclude: Style check
- [ ]  Exclude: Fast test
- [ ]  Exclude: All with ASAN
- [ ]  Exclude: All with TSAN, MSAN, UBSAN, Coverage
- [ ]  Exclude: All with aarch64, release, debug
---
- [ ]  Run only fuzzers related jobs (libFuzzer fuzzers, AST fuzzers, etc.)
- [ ]  Exclude: AST fuzzers
---
- [ ]  Do not test
- [ ]  Woolen Wolfdog
- [ ]  Upload binaries for special builds
- [ ]  Disable merge-commit
- [ ]  Disable CI cache

### CODE REVIEW

**Comment by robot-ch-test-poll1** [comment]
*This is an automated comment for commit 8b81862bbf698de3ce548ecd270e7dd2b1f9e71a with description of existing statuses. It's updated for the latest CI running*

[✅ Click here](https://s3.amazonaws.com/clickhouse-test-reports/72510/8b81862bbf698de3ce548ecd270e7dd2b1f9e71a/ci_running.html) to open a full report in a separate page

<details><summary>Successful checks</summary>
<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>Style check</td><td>Runs a set of checks to keep the code style clean. If some of tests failed, see the related log from the report</td><td><a href="https://s3.amazonaws.com/clickhouse-test-reports/72510/8b81862bbf698de3ce548ecd270e7dd2b1f9e71a/style_check.html">✅ success</a></td></tr>
<tbody>
</table>
</details>

File: tests/queries/0_stateless/03279_pr_3_way_joins.reference
Comment (nickitat) [inline tests/queries/0_stateless/03279_pr_3_way_joins.reference:1]: could you make it a bit smaller pls

File: src/Planner/PlannerJoinTree.cpp [hunk 1]
```diff
@@ -1985,27 +1987,51 @@ JoinTreeQueryPlan buildJoinTreeQueryPlan(const QueryTreeNodePtr & query_node,
...
+    /// disable parallel replicas for n-way join with FULL JOIN involved
+    if (joins_count > 1 && is_full_join)
+        planner_context->getMutableQueryContext()->setSetting("enable_parallel_replicas", Field{0});
+
     /** If left most table expression query plan is planned to stage that is not equal to fetch columns,
       * then left most table expression is responsible for providing valid JOIN TREE part of final query plan.
       *
       * Examples: Distributed, LiveView, Merge storages.
       */
+    // find parent node in the table expressions stack
```
Comment (nickitat) [inline src/Planner/PlannerJoinTree.cpp:2022]: imo it is pretty obvious that we're searching for the parent node in this code, would be more valuable to explain why we're doing this

**Review by nickitat — APPROVED** [review]

**Comment by nickitat** [comment]
you could also consider as a replacement to comparing whole output
* calculate sum(cityHash(*)) of all rows
* save only first N rows

**Pull request merged**